src/test/txpackage_tests.cpp

```diff
@@ -0,0 +1,566 @@
+        CTransactionRef tx_parent_invalid{MakeTransactionRef(mtx_parent_invalid)};
+        Package package_invalid_parent{tx_parent_invalid, tx_child};
+        submitPool.clear();
+        submitPool.reserve(package_invalid_parent.size());
+        auto result_quit_early = TestPackageAcceptance(package_invalid_parent, state, packageState, opt);
+        BOOST_CHECK(!result_quit_early);
+        BOOST_CHECK(state.IsValid());
+        BOOST_CHECK_EQUAL(state.GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(state.GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(packageState[tx_parent_invalid->GetHashMalFix()].GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(packageState[tx_parent_invalid->GetHashMalFix()].GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(packageState[tx_parent_invalid->GetHashMalFix()].missingInputs, true);
+        BOOST_CHECK_EQUAL(packageState[tx_child->GetHashMalFix()].GetRejectCode(), REJECT_DUPLICATE);
+        BOOST_CHECK_EQUAL(packageState[tx_child->GetHashMalFix()].GetRejectReason(), "txn-already-in-mempool");
+        BOOST_CHECK_EQUAL(submitPool.size(), 0);
+
+        auto result_unrelated_submit = SubmitPackageToMempool(submitPool, stateSubmit);
+
+        BOOST_CHECK(result_unrelated_submit);
+        BOOST_CHECK_EQUAL(stateSubmit.GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(stateSubmit.GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(mempool.size(), expected_pool_size + 13);
+    }
+
+    // high fee tx.
+    CMutableTransaction tx_child2{*tx_child};
+    tx_child2.vin.push_back(CTxIn(COutPoint(package_unrelated[0]->GetHashMalFix(), 0), CScript()));
+    Package package_missing_parent;
+    package_missing_parent.push_back(tx_parent);
+    package_missing_parent.push_back(MakeTransactionRef(tx_child2));
+    {
+        submitPool.clear();
+        submitPool.reserve(package_missing_parent.size());
+        const auto result_missing_parent = TestPackageAcceptance(package_missing_parent, state, packageState, opt);
+        BOOST_CHECK(!result_missing_parent);
+        BOOST_CHECK(state.IsValid());
+        BOOST_CHECK_EQUAL(state.GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(state.GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(packageState[tx_parent->GetHashMalFix()].GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(packageState[tx_parent->GetHashMalFix()].GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(packageState[tx_child2.GetHashMalFix()].GetRejectCode(), REJECT_HIGHFEE);
+        BOOST_CHECK_EQUAL(packageState[tx_child2.GetHashMalFix()].GetRejectReason(), "absurdly-high-fee");
+        BOOST_CHECK_EQUAL(submitPool.size(), 0);
+
+        auto result_unrelated_submit = SubmitPackageToMempool(submitPool, stateSubmit);
+        BOOST_CHECK(result_unrelated_submit);
+        BOOST_CHECK_EQUAL(stateSubmit.GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(stateSubmit.GetRejectReason(), "");
+        BOOST_CHECK_EQUAL(mempool.size(), expected_pool_size + 13);
+    }
+
```

Comment:
There doesn't seem to be a test for:

https://github.com/bitcoin/bitcoin/blob/master/src/test/txpackage_tests.cpp#L479-L493

Reply:
This check is not similar to bitcoin. It is included in test `package_validation_tests`.  Our package creation and validation checks are a little different from bitcoin, the call to `FilterMempoolDuplicates` filters the mempool transaction before the package creation. So all packages are expected to be deduplicated.

Reply:
`package_validation_tests` has two cases of tests: Cases where the parent and child are in mempool and cases where the parent and child are not in mempool.

Is there a test for behavior when only the parent is already in mempool? And only the child is in the orphan pool.

Reply:
oh! I understand the exact case only now. There is no test for such a scenario. I'll add it.